### PR TITLE
🏞 `MMIF` source 

### LINKED ISSUE #151: Source MMIF
As a Clammer, I can select the `source_mmif` to be used for a Media File in a Batch.

Before starting a Batch, I can choose to:
- Use the existing `source_mmif` if, available (default)
- Use a "blank" MMIF
  - ("new", "empty", "clean"?)
- Select a `source_mmif` for a MediaFile from a list of available MMIF
- Select from a list of available MMIF from a specific Pipeline

Optionally, there can be other selection methods:
- `n`th Most recent
- `n`th Oldest
- Matching string query

```[tasklist]
### Tasks
- [x] Add `MMIF` table
- [x] Update `MediaFile` model with `source_mmif`, `mmifs`
- [ ] https://github.com/WGBH-MLA/chowda/issues/165
- [ ] https://github.com/WGBH-MLA/chowda/pull/168
- [ ] https://github.com/WGBH-MLA/chowda/issues/174
- [x] Add `input_mmifs` to `Batch`
- [ ] https://github.com/WGBH-MLA/chowda/issues/166
- [ ] Select `source_mmif` from `Pipeline`
```

### LINKED ISSUE #165: Allow "blank" MMIF Source as an option for Batches
### Summary

As a Clammer
When I am creating a Batch
And when the "MMIF Source" field is left blank
And when the Batch is started
Then the Batch will run it's associated Media Files through it's associated Pipeline using "blank" MMIF as input
In order to get Pipeline results without an input MMIF.

### Related

_No response_

### Alternatives

_No response_

### Additional context

_No response_
```[tasklist]
### Tasks
- [x] Add "new MMIF" checkbox
- [x] Parse form in Chowda
- [x] Update logic in Pipeline
```

### LINKED ISSUE #166: Allow other Pipelines for a Batch's MMIF Source
### Summary

As a Clammer
When editing a Batch that has not started
And when I select an existing Pipeline for the "MMIF Source" field
And when I start the Batch
Then the input MMIF for each Media File within the Batch will be the most recent MMIF found for the MediaFile+Pipeline combination -- i.e. the "canonical" MMIF for that Media File in that Pipeline
In order to do a Pipeline Run using previously generated MMIF as input.

### Related

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### LINKED ISSUE #174: Migrate relationship: 1 MediaFile => Many SonyCiAssets
# Currently
Many `SonyCiAssets` <=> Many `MediaFiles`

# Should be
One `MediaFile` => Many `SonyCiAssets`

### LINKED ISSUE #175: Ingest only `audio`, `video`
# Ingest
Currently, we are ingesting all SonyCi `assets` that match `^cpb-aacip-*`.

Filtering by `type:` `audio` or `video`  will exclude many of the duplicate assets in in SonyCi (hash files, xmls).

### PR DESCRIPTION
**Pull request by mrharpo**
# MMIF
Adds `MMIF` table
- `media_file`
- `batch_inputs: List[Batch]`
- `batch_output: Batch`
- `metaflow_run`
- `mmif_location`
  - s3 key to mmif

## Batch `input_mmifs`
- Creates `input_mmifs: List[MMIF]`
- Creates `output_mmifs: List[MMIF]`

## Ingest
- Updates ingest for new models
- Filters `media_files` with `type == 'Video' or 'Audio'` 

### New MMIF
Adds a checkbox input to the Start Batch form: "Start from blank MMIF?"

### MediaType
- Changes `MediaType` Enum to have value match attribute name
- Updated manually in migration file

With https://github.com/WGBH-MLA/mario/pull/17

Closes #165, Closes #151, Closes #174, Closes #175, Closes #166

### CODE REVIEW

**Review by afred — APPROVED** [review]
👍

File: chowda/models.py [hunk 1]
```diff
@@ -271,8 +296,58 @@ class MetaflowRun(SQLModel, table=True):
...
+            "primaryjoin": "MMIF.batch_output_id==Batch.id",
+        },
+    )
+    batch_inputs: List[Batch] = Relationship(
+        back_populates='input_mmifs',
+        link_model=MMIFBatchInputLink,
+    )
+
+    mmif_json: Dict[str, Any] = Field(sa_column=Column(JSON), default=None)
+    mmif_location: Optional[str] = Field(default=None)
```
[the remark this answers is not in the document]
Comment (mrharpo) [reply chowda/models.py:346]: We don't need it anymore. I wanted to make sure we agreed we didn't need it before removing it.

Also, after our database scaling issues on Friday, I'm more convinced we don't want to cache the MMIF here.

[the remark this answers is not in the document]
Comment (afred) [reply chowda/models.py:346]: 👍 unneeded

**Review by afred — APPROVED** [review]
👍

**Pull request merged**